plugin-api/src/main/java/org/hyperledger/besu/plugin/services/SecurityModuleService.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.services;
+
+import org.hyperledger.besu.plugin.Unstable;
+import org.hyperledger.besu.plugin.services.securitymodule.SecurityModule;
+
+import java.util.Optional;
+import java.util.function.Function;
+
+/**
+ * This service allows plugins to register Security Module which is abstraction of cryptographic
+ * operations by deferring to specific provider such as BouncyCastle
+ */
+@Unstable
+public interface SecurityModuleService {
+
+  @FunctionalInterface
+  interface SecurityModuleProvider extends Function<BesuConfiguration, SecurityModule> {
```

Comment:
Is there any benefit for this to extend Function? Would of expected this to be a top level class with a create method following the style done for storage API which has a KeyValueStorageFactory

Reply:
refactored to top level. The only benefit is adhering to existing functional interface. We can technically avoid declaring apply method altogether.